include/boost/beast/experimental/core/uri/rfc3986.hpp

```diff
@@ -0,0 +1,96 @@
+//
+// Copyright (c) 2016-2018 Vinnie Falco (vinnie dot falco at gmail dot com)
+// Copyright (c) 2018 oneiric (oneiric at i2pmail dot org)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+#ifndef BOOST_BEAST_CORE_URI_DETAIL_RFC3986_HPP
+#define BOOST_BEAST_CORE_URI_DETAIL_RFC3986_HPP
+
+namespace boost {
+namespace beast {
+namespace uri {
+
+bool is_alpha(char c) {
+  unsigned constexpr a = 'a';
+  return ((static_cast<unsigned>(c) | 32) - a) < 26U;
+}
+
+bool is_digit(char c) {
+  unsigned constexpr zero = '0';
+  return (static_cast<unsigned>(c) - zero) < 10;
+}
+
+bool is_hex(char c) {
+  return (c == 'a' || c == 'b' || c == 'c' || c == 'd' || c == 'e' ||
+          c == 'f' || c == 'A' || c == 'B' || c == 'C' || c == 'D' ||
+          c == 'E' || c == 'F' || is_digit(c));
+}
+
```

Comment:
Did you intend for the above three functions to be inline? (or perhaps even constexpr)